src/prefect/orion/schemas/responses.py

```diff
@@ -129,3 +139,52 @@ class Config:
     worker_pool_id: UUID
     worker_pool_queue_id: UUID
     flow_run: schemas.core.FlowRun
+
+
+@copy_model_fields
+class DeploymentResponse(ORMBaseModel):
+    name: str = FieldFrom(schemas.core.Deployment)
+    version: Optional[str] = FieldFrom(schemas.core.Deployment)
+    description: Optional[str] = FieldFrom(schemas.core.Deployment)
+    flow_id: UUID = FieldFrom(schemas.core.Deployment)
+    schedule: Optional[schemas.schedules.SCHEDULE_TYPES] = FieldFrom(
+        schemas.core.Deployment
+    )
+    is_schedule_active: bool = FieldFrom(schemas.core.Deployment)
+    infra_overrides: Dict[str, Any] = FieldFrom(schemas.core.Deployment)
+    parameters: Dict[str, Any] = FieldFrom(schemas.core.Deployment)
+    tags: List[str] = FieldFrom(schemas.core.Deployment)
+    work_queue_name: Optional[str] = FieldFrom(schemas.core.Deployment)
+    parameter_openapi_schema: Optional[Dict[str, Any]] = FieldFrom(
+        schemas.core.Deployment
+    )
+    path: Optional[str] = FieldFrom(schemas.core.Deployment)
+    entrypoint: Optional[str] = FieldFrom(schemas.core.Deployment)
+    manifest_path: Optional[str] = FieldFrom(schemas.core.Deployment)
+    storage_document_id: Optional[UUID] = FieldFrom(schemas.core.Deployment)
+    infrastructure_document_id: Optional[UUID] = FieldFrom(schemas.core.Deployment)
+    created_by: Optional[CreatedBy] = FieldFrom(schemas.core.Deployment)
+    updated_by: Optional[UpdatedBy] = FieldFrom(schemas.core.Deployment)
+    worker_pool_name: Optional[str] = Field(
+        default=None,
+        description="The name of the deployment's worker pool.",
+        example="my-worker-pool",
+    )
+    worker_pool_queue_name: Optional[str] = Field(
+        default=None,
+        description="The name of the deployment's worker pool queue.",
+        example="my-worker-pool-queue",
+    )
+
+    @classmethod
+    def from_orm(
+        cls, orm_deployment: "prefect.orion.database.orm_models.ORMDeployment"
+    ):
+        response = super().from_orm(orm_deployment)
+        if orm_deployment.worker_pool_queue:
```

Comment:
same here - shouldn't this logic be flipped to check on `orm_deployment.worker_pool` and then optionally tack on `worker_pool_queue` within the conditional?

Reply:
A deployment will be related to a  worker pool queue which is why there's a check for `worker_pool_queue`. The name for th worker pool comes from the relationship between the queue and pool. It's useful to have both names so that they can be explicitly displayed to the user instead of needing to understand that if there is no queue name then the deployment schedules work on the default queue of the pool.